File: language/java_study/javaLearn/src/main/java/com/iamhotcenplotz/chapter14/exercises/Exercise05_.java
package com.iamhotcenplotz.chapter14.exercises;

import java.util.TreeSet;

/**
 * @author Der Hotcenplotz
 * @version 1.0
 */
public class Exercise05_ {
    public static void main(String[] args) {
        TreeSet treeSet = new TreeSet();

        // 因为Person 没有实现Comparable接口，所以这里会报错。
        treeSet.add(new Person());
    }
}

class Person implements Comparable<Person> {

    @Override
    public int compareTo(Person o) {
        return this.toString().compareTo(o.toString());
    }

    @Override
    public String toString() {
        return "";
    }
}

File: language/scala_study/ScalaLearn/src/main/scala/com/hotcenplotz/chapter14/exercise_/Exercise05_.scala
package com.hotcenplotz.chapter14.exercise_

/**
 * @author: Der Hotcenplotz
 * @version： 1.0 
 * @date: 2025-05-26 21:42 
 * @description:
 * */
object Exercise05_ {
    def main(args: Array[String]): Unit = {
        println(values(x => x + 100, 10, 20))
        println(values(x => x * 2, 10, 20))
        println(values(x => x * x, 10, 20))
        println(values(x => x, 10, 20))
    }
    
    def values(func: Int => Int, low: Int, high: Int): List[(Int, Int)] = {
        var newList: List[(Int, Int)] = Nil
        (low to high).foreach { x =>
            newList = (x, func(x)) :: newList
        }
        newList.reverse
    }
}
